cmd/postgres-operator/main.go

```diff
@@ -58,13 +60,40 @@ func initLogging() {
 }
 
 //+kubebuilder:rbac:groups="coordination.k8s.io",resources="leases",verbs={get,create,update,watch}
+//+kubebuilder:rbac:groups="authentication.k8s.io",resources="tokenreviews",verbs={create}
+//+kubebuilder:rbac:groups="authorization.k8s.io",resources="subjectaccessreviews",verbs={create}
 
 func initManager(ctx context.Context) (runtime.Options, error) {
 	log := logging.FromContext(ctx)
 
 	options := runtime.Options{}
 	options.Cache.SyncPeriod = initialize.Pointer(time.Hour)
 
+	// If we aren't using it, http/2 should be disabled
+	// due to its vulnerabilities. More specifically, disabling http/2 will
+	// prevent from being vulnerable to the HTTP/2 Stream Cancellation and
+	// Rapid Reset CVEs. For more information see:
+	// - https://github.com/advisories/GHSA-qppj-fm5r-hxr3
+	// - https://github.com/advisories/GHSA-4374-p667-p6c8
+	disableHTTP2 := func(c *tls.Config) {
+		log.Info("disabling http/2")
+		c.NextProtos = []string{"http/1.1"}
+	}
+	var tlsOpts []func(*tls.Config)
+	tlsOpts = append(tlsOpts, disableHTTP2)
+	options.Metrics.TLSOpts = tlsOpts
```

Comment:
⛏️ I see this follows the Kubebuilder v4 template, but we can do it with fewer types and lines.

🤔 Should we include the word "metrics" or so in the message? How about an active/positive phrasing?

```suggestion
	options.Metrics.TLSOpts = append(options.Metrics.TLSOpts, func(c *tls.Config) {
		log.Info("enabling metrics via http/1.1")
		c.NextProtos = []string{"http/1.1"}
	})
```